src/view/treeNodes/fileChangeNode.ts

```diff
@@ -98,7 +101,7 @@ export class FileChangeNode extends TreeNode implements vscode.TreeItem {
 	}
 
 	constructor(
-		public readonly parent: TreeNodeParent,
+		public parent: TreeNodeParent,
```

Comment:
I see why you needed to remove the `readonly` to change the `parent` later. To make the code more self-documenting, can we:
- Make `parent` `_parent` and make it private
- Have a `get parent`
- Have a public method for setting the parent. Something like `updateParentDirectory`.

Reply:
I believe this is not possible as we require the parent field due to extending TreeNode.
Because of this, it won't allow us to create accessors (get parent).
We also can't restrict the visibility of parent to private as the super class has it has public.